File: booksamples/chapter-01/src/main/java/org/vulkanb/eng/IAppLogic.java
package org.vulkanb.eng;

import org.vulkanb.eng.graph.Render;
import org.vulkanb.eng.scene.Scene;

public interface IAppLogic {

    void cleanup();

    void input(Window window, Scene scene, long diffTimeMillis);

    void init(Window window, Scene scene, Render render);

    void update(Window window, Scene scene, long diffTimeMillis);
}

File: booksamples/chapter-05/kotlin/src/main/kotlin/org/vulkanb/eng/IAppLogic.kt
package org.vulkanb.eng

import org.vulkanb.eng.graph.Render
import org.vulkanb.eng.scene.Scene

interface IAppLogic {
    fun cleanup()
    fun input(window: Window?, scene: Scene?, diffTimeMillis: Long)
    fun init(window: Window?, scene: Scene?, render: Render?)
    fun update(window: Window?, scene: Scene?, diffTimeMillis: Long)
}
